### PR TITLE
build: Add Dockerfile & Dockerignore

### PR DESCRIPTION
**Pull request by PCOffline**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM node:14
```
Comment (ShragaUser) [inline Dockerfile:1]: The base Image should be an "alpine" typed one

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+WORKDIR /usr/src/app
+COPY ["package.json", "package-lock.json", "./"]
+RUN npm install --silent
+COPY . .
+
+CMD ["npm", "run", "dev"]
```
Comment (ShragaUser) [inline Dockerfile:7]: Dockerfile should be production ready therefore usage of development mom script is wrong here.
Consider creating a dev.Dockerfile for development etc'

Comment (yonatandt) [inline Dockerfile:7]: `npm run dev`?
This should run in production, it is not the correct script to run.

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+node_modules
+dist
```
Comment (ShragaUser) [inline .dockerignore:2]: Environment files should also be ignored 
Be sure to add .env files and the like to docker ignore
Also IDE specific files

**Review by yonatandt — CHANGES_REQUESTED** [review]
The image could be lighter than this.

**Review by ShragaUser — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,10 @@
+FROM node:alpine
```
Comment (ShragaUser) [inline Dockerfile:1]: node version should be explicit.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+WORKDIR /usr/src/app
+COPY ["package.json", "package-lock.json", "./"]
+RUN npm install --silent
```
Comment (ShragaUser) [inline Dockerfile:4]: Should probably have production tag
No need for dev dependencies

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+COPY . .
```
Comment (ShragaUser) [inline Dockerfile:5]: FYI
proper production docker files usually have multi-layer build 
Where only the dist is copied to a new layer and then run over nodejs without build dependencies. 

Not relevant in our case, but if you'd like a challenge, try to think about why this is the case and what can be done to improve this.

**Review by ShragaUser — CHANGES_REQUESTED** [review]

**Comment by yonatandt** [comment]
@PCOffline Make sure you sign your commits.

**Comment by PCOffline** [comment]
@yonatandt The node:14-alpine is already there. Probably a problem with File Changes

**Comment by yonatandt** [comment]
@PCOffline Also in dev.Dockerfile?

**Pull request merged**